### PR TITLE
Packetizer Version 2 support

### PR DESCRIPTION
**Pull request by slacrherbst**

### CODE REVIEW

**Comment by ruck314** [comment]
@slacrherbst: Has the software been updated with the latest updates in the firmware Version 2 packetizer?

**Comment by slacrherbst** [comment]
Yes its been sitting in a pull request for a couple of weeks.

**Comment by ruck314** [comment]
The firmware formatting has recently changed.  Does the software reflect these firmware changes?  If yes, then I can start to test this software on the dev board platform.

**Comment by slacrherbst** [comment]
If you mean the version number addition, yes I updated to match that.

**Comment by ruck314** [comment]
Ok, I will try it on test it on the dev board platform now

**Comment by ruck314** [comment]
@slacrherbst: My existing code is this:
```
    ethLink = pr.protocols.UdpRssiPack(
        host = args.ip,
        port = 8192,
        size = 1400,
        )
```
What do I need to change to select whether I am using packetizer V1 or packeter V2?

**Comment by slacrherbst** [comment]
Added a new arg packVer to the wrapper.

**Comment by ruck314** [comment]
@slacrherbst:

**Comment by ruck314** [comment]
@slacrherbst: Is "75df5ff" a bug fix for the issue that I am seeing?

**Comment by slacrherbst** [comment]
No. There must be a protocol mismatch someplace. Let me add debugging.

**Comment by ruck314** [comment]
@slacrherbst: Since the RSSI already has a checksum, we have the option to turn on/off the CRC in the packeterizer.  For high bandwidth performance, should we disable the packeter CRC checking/generation? Or should we assume always CRC in the packerize for RSSI+UDP communication?  I think we should always do one way or the other (and not both).  Having support support two different modes of CRC will get confusing fast.

**Comment by ruck314** [comment]
I would vote to "always" do the CRC in the packetizer (another level of error checking).

**Comment by slacrherbst** [comment]
Agreed.

**Comment by slacrherbst** [comment]
To Enable packetizer debugging:
rogue.Logging.setFilter("packetizer.Controller",rogue.Logging.Debug)

**Comment by slacrherbst** [comment]
I just pushed a version with more debugging. See previous comment to enable debug.

**Comment by ruck314** [comment]
@slacrherbst: After these modification in the SURF pull request (https://github.com/slaclab/surf/pull/178), I am now able to do register transaction. 
I am now testing interleaved traffic (register + PRBS TX).  I have observed that if the PRBS TX packet is larger than the chucking size that PRBS TX packets are getting dropped.

**Comment by slacrherbst** [comment]
This is the software side sending packets that are too large?

What did you set for the size value on the software side?

**Comment by slacrherbst** [comment]
Can you point me to the test script?

**Comment by ruck314** [comment]
Here's the image:

**Comment by slacrherbst** [comment]
Ok. so you are transmitting from firmware to software? This has to be a firmware error then? I thought you were talking about a firmware transmit.

**Comment by ruck314** [comment]
Yes, firmware-to-software transfer.
I have not tested the PRBS software-to-firmware path yet

**Comment by slacrherbst** [comment]
Nothing gets received in the software for the dest? No Debug? It could still be a software error

**Comment by slacrherbst** [comment]
Please remove setup_env.csh from the commit.

**Comment by ruck314** [comment]
Done.

**Comment by ruck314** [comment]
@slacrherbst: https://github.com/slaclab/rogue/blob/ESROGUE-153/src/rogue/protocols/packetizer/ControllerV2.cpp#L142

**Comment by slacrherbst** [comment]
Fixed it. Bad merge of two conditions.

**Comment by ruck314** [comment]
@slacrherbst: can you check the thread locking of module?  If I have polling turned off, I don't get any drop packets or PRBS errors.  Then when I hit the ReadAll button a bunch of times, I get CRC errors in the packetizer and PRBS errors.

**Comment by slacrherbst** [comment]
Locking looks ok. CRC errors in the packetizer are interesting. That means the packet got corrupted or shortened someplace between the firmware packetizer and the software module.

**Comment by ruck314** [comment]
@slacrherbst: I think that we should wait for Ben to complete PGPv3 simulation regression testing to confirm that FW-to-FW still looks good.

**Comment by slacrherbst** [comment]
Once you get this moving data we should merge it into pre-release.

**Comment by ruck314** [comment]
@slacrherbst: I need to test the path where the software is the PRBS TX and the firmware is PRBS RX.  Can you point me to an example of this?

**Comment by slacrherbst** [comment]
prbTx = pyrogue.utilities.prbs.PrbsTx(name="prbTx")
pyrogue.streamConnect(prbTx, txVc)

You will be able to start/stop it in software.

**Comment by ruck314** [comment]
Completed by software testing of this software branch with respect to the SURF (https://github.com/slaclab/surf/pull/178/commits/5e3942cee2ed28f96e222bfca68320bf09e41c80).  I have tested PRBS in both directions for both single chucked up frames and multiple chucked up frames.  I think we are ready for the merge to pre-release branch.
@slacrherbst: Anything else that you want to test before the merge?  Do you want to resolve the merge conflict?

**Comment by slacrherbst** [comment]
We will still want full CRC mode over Ethernet when going firmware -> firmware, but we should not with software if we want performance. So configurable.

**Comment by ruck314** [comment]
Unless there is going to be a change to the protocol definition, I think that we are ready to merge pull request.

File: src/rogue/protocols/packetizer/Controller.cpp [hunk 1]
```diff
@@ -55,7 +55,8 @@ rpp::Controller::Controller ( uint32_t segmentSize, rpp::TransportPtr tran, rpp:
 
    headSize_ = headSize;
    tailSize_ = tailSize;
-   enCrc_    = false; 
+   enIbCrc_ = false; // Need to bring this up to the top level
```
Comment (ruck314) [inline src/rogue/protocols/packetizer/Controller.cpp]: @slacrherbst: Can I have you add the hooks to map this to the higher levels?

**Comment by slacrherbst** [comment]
Ok. Will do.

File: src/rogue/protocols/packetizer/Controller.cpp [hunk 2]
```diff
@@ -55,8 +55,6 @@ rpp::Controller::Controller ( uint32_t segmentSize, rpp::TransportPtr tran, rpp:
 
    headSize_ = headSize;
    tailSize_ = tailSize;
-   enIbCrc_ = false; // Need to bring this up to the top level
```
Comment (ruck314) [inline src/rogue/protocols/packetizer/Controller.cpp]: @slacrherbst: Where does the (enIbCrc_ = false) and (enObCrc_ = true) as default in the higher level software?

**Pull request merged**